src/test/java/javaslang/collection/euler/ProjectEulerTest.java

```diff
@@ -85,9 +84,37 @@ private static long fibonacci(int order) {
      * <p>See also <a href="https://projecteuler.net/problem=7">projecteuler.net problem 7</a>.</p>
      */
     @Test
-    @Ignore
-    public void shouldSolveProblem7() {
-        final int actual = Primes.asStream().drop(10_000).head();
-        assertThat(actual).isEqualTo(0); // TODO, see Primes#sieve(Stream)
+    public void test7() {
+        assertThat(primeNo(1)).isEqualTo(2);
+        assertThat(primeNo(2)).isEqualTo(3);
+        assertThat(primeNo(3)).isEqualTo(5);
+        assertThat(primeNo(4)).isEqualTo(7);
+        assertThat(primeNo(5)).isEqualTo(11);
+        assertThat(primeNo(6)).isEqualTo(13);
+        assertThat(primeNo(10_001)).isEqualTo(104_743);
+    }
+
+    private static int primeNo(int index) {
+        return Match
+                .when(1, (i) -> 2)
+                .orElse(() -> knownPrimes.drop(index - 1).head())
+                .apply(index);
+    }
+
+    private static final Stream<Integer> knownPrimes = Stream.gen(2, (p) -> nextPrime(p));
+
+    private static int nextPrime(int previousPrime) {
+        final Integer nextPrime = Match
```

Comment:
`Match` is good for recognizing objects of a specific type. I would suggest to use the good old `if` instead, also because Match uses deserialization of lambdas which costs cycles.

```java
if (previousPrime == 2) {
    return 3;
} else {
    return Stream.gen(previousPrime + 2, v -> v + 2)
            .filter(i -> !isEvenlyDiversableByKnownPrimes(previousPrime, i))
            .take(1).head();
}
```

or even

```java
return (previousPrime == 2) ? 3 : Stream.gen(previousPrime + 2, v -> v + 2)
            .filter(i -> !isEvenlyDiversableByKnownPrimes(previousPrime, i))
            .take(1).head();
```

This should be much faster.